vendor/dune

```diff
@@ -1,55 +1,171 @@
-; There are 4 configrations for the blake3 rules:
+; There are 3 configrations for the blake3 rules:
 ; 1. Handwritten assembly for x86_64
 ; 2. SIMD instructions with neon intrinsics for arm64
-; 3. SIMD instructions with x86 intrinsics for amd64
-; 4. Portable C implementation
+; 3. Portable C implementation
 
-; We only handle the last 3 configurations here.
-
-; x86 intrinsics
+; 1. Handwritten assembly for x86_64
 
 (foreign_library
  (enabled_if
   (= %{architecture} "amd64"))
  (archive_name blake3)
  (language c)
- (flags :standard -O2)
  (names blake3 blake3_portable blake3_dispatch)
- (extra_objects blake3_avx2 blake3_avx512 blake3_sse2 blake3_sse41))
+ (extra_objects
+  blake3_avx2_x86-64
+  blake3_avx512_x86-64
+  blake3_sse2_x86-64
+  blake3_sse41_x86-64))
+
+; 1.1: Unix assembly for x86_64
 
 (rule
  (enabled_if
-  (= %{architecture} "amd64"))
- (deps blake3_avx2.c)
- (targets blake3_avx2.o)
+  (and
+   (= %{architecture} "amd64")
+   (or
+    (= %{system} "linux")
+    (= %{system} "macosx"))))
+ (deps blake3_avx2_x86-64_unix.S)
+ (targets blake3_avx2_x86-64.o)
  (action
-  (run %{cc} -mavx2 -c %{deps} -o %{targets})))
+  (run %{cc} -c %{deps} -o %{targets})))
 
 (rule
  (enabled_if
-  (= %{architecture} "amd64"))
- (deps blake3_avx512.c)
- (targets blake3_avx512.o)
+  (and
+   (= %{architecture} "amd64")
+   (or
+    (= %{system} "linux")
+    (= %{system} "macosx"))))
+ (deps blake3_avx512_x86-64_unix.S)
+ (targets blake3_avx512_x86-64.o)
  (action
-  (run %{cc} -mavx512f -mavx512vl -mavx512bw -c %{deps} -o %{targets})))
+  (run %{cc} -c %{deps} -o %{targets})))
 
 (rule
  (enabled_if
-  (= %{architecture} "amd64"))
- (deps blake3_sse2.c)
- (targets blake3_sse2.o)
+  (and
+   (= %{architecture} "amd64")
+   (or
+    (= %{system} "linux")
+    (= %{system} "macosx"))))
+ (deps blake3_sse2_x86-64_unix.S)
+ (targets blake3_sse2_x86-64.o)
  (action
-  (run %{cc} -msse2 -c %{deps} -o %{targets})))
+  (run %{cc} -c %{deps} -o %{targets})))
 
 (rule
  (enabled_if
-  (= %{architecture} "amd64"))
- (deps blake3_sse41.c)
- (targets blake3_sse41.o)
+  (and
+   (= %{architecture} "amd64")
+   (or
+    (= %{system} "linux")
+    (= %{system} "macosx"))))
+ (deps blake3_sse41_x86-64_unix.S)
+ (targets blake3_sse41_x86-64.o)
```

Comment:
Should be `%{ext_obj}`

Reply:
Out of interest does it ever change on unix?

Reply:
Not that I know of, but I still think we should use this variable because it costs us nothing and is easy to break things on windows if we don't